src/Parser/Multipart.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace React\Http\Parser;
+
+use React\Http\File;
+use React\Http\Request;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\ThroughStream;
+
+/**
+ * Parse a multipart body
+ *
+ * Original source is from https://gist.github.com/jas-/5c3fdc26fedd11cb9fb5
+ *
+ * @author jason.gerfen@gmail.com
+ * @author stephane.goetz@onigoetz.ch
+ * @license http://www.gnu.org/licenses/gpl.html GPL License 3
```

Comment:
re-write or use from #13. We can't mix licences and can't steal code that is licensed under GPL.